src/iden3comm/packers/anon-crypt.ts

```diff
@@ -0,0 +1,243 @@
+        const pkStore = await this._kms.getKeyProvider(keyId.type)?.getPkStore();
+        if (!pkStore) {
+          throw new Error(`Key provider not found for ${keyId.type}`);
+        }
+        try {
+          return JSON.parse(await pkStore.get({ alias: keyId.id })) as CryptoKey;
+        } catch (error) {
+          throw new Error(`Key not found for ${keyId.id}`);
+        }
+      })
+    );
+
+    const result = await Promise.any(promises);
+
+    const { plaintext } = result;
+
+    return JSON.parse(byteDecoder.decode(plaintext)) as BasicMessage;
+  }
+
+  mediaType(): MediaType {
+    return MediaType.EncryptedMessage;
+  }
+
+  /** {@inheritDoc IPacker.getSupportedProfiles} */
+  getSupportedProfiles(): string[] {
+    return this._supportedProtocolVersions.map(
+      (v) => `${v};env=${this.mediaType()};alg=${this._supportedAlgorithms.join(',')}`
+    );
+  }
+
+  /** {@inheritDoc IPacker.isProfileSupported} */
+  isProfileSupported(profile: string): boolean {
+    const { protocolVersion, env, circuits, alg } = parseAcceptProfile(profile);
+
+    if (!this._supportedProtocolVersions.includes(protocolVersion)) {
+      return false;
+    }
+    if (env !== this.mediaType()) {
+      return false;
+    }
+
+    if (circuits) {
+      throw new Error(`Circuits are not supported for ${env} media type`);
+    }
+
+    return Boolean(
+      alg?.some((a) => this._supportedAlgorithms.includes(a as AcceptJweKEKAlgorithms))
+    );
+  }
+
+  extractPublicKeyBytes = (vm: VerificationMethod): JsonWebKey | Uint8Array | null => {
+    if (vm.publicKeyBase58) {
+      return base58ToBytes(vm.publicKeyBase58);
+    }
+    if (vm.publicKeyBase64) {
+      return base64UrlToBytes(vm.publicKeyBase64);
+    }
+    if (vm.publicKeyHex) {
+      return hexToBytes(vm.publicKeyHex);
+    }
+    if (vm.publicKeyJwk) {
+      return vm.publicKeyJwk;
+    }
+
+    return null;
+  };
+
+  private async getRecipientsJWKs(recipients: RecipientInfo[]): Promise<
+    {
+      alg: string;
+      did: string;
+      keyType: VerificationMethodType;
+      kid: string;
+      recipientJWK: JsonWebKey;
+    }[]
+  > {
+    return Promise.all(
+      recipients.map(async (recipient) => {
+        if (!recipient.did) {
+          throw new Error('Missing target key id');
+        }
+        const recipientDidDoc: DIDDocument | null =
+          recipient.didDocument ??
+          (await this._documentResolver.resolve(recipient.did.string()))?.didDocument;
+
+        if (!recipientDidDoc) {
+          throw new Error('Recipient DID document not found');
+        }
+
+        const vms = resolveVerificationMethods(recipientDidDoc);
+
+        if (!vms.length) {
+          throw new Error(
+            `No verification methods defined in the DID document of ${recipientDidDoc.id}`
+          );
+        }
+
+        const keyType = recipient.keyType ?? 'JsonWebKey2020';
+        const alg = recipient.alg ?? AcceptJweKEKAlgorithms.RSA_OAEP_256;
```

Comment:
Illia is using jwkKeyType and jwkKeyAlg filters

Reply:
so what is the point?